sdk/src/test/java/com/mapbox/maps/renderer/MapboxRenderThreadTest.kt

```diff
@@ -119,80 +170,72 @@ class MapboxRenderThreadTest {
 
   @Test
   fun onSurfaceSizeChangedSameSizeTest() {
-    val latch = CountDownLatch(1)
-    every { mapboxRenderer.createRenderer() } answers { latch.countDown() }
-    mockValidSurface()
-    Shadows.shadowOf(renderHandlerThread.handler?.looper).pause()
+    initRenderThread()
+    provideValidSurface()
+    pauseHandler()
     mapboxRenderThread.onSurfaceSizeChanged(1, 1)
-    Shadows.shadowOf(renderHandlerThread.handler?.looper).idle()
-    if (!latch.await(waitTime, TimeUnit.MILLISECONDS)) {
-      throw TimeoutException()
-    }
-    verify {
+    idleHandler()
+    verifyOnce {
       mapboxRenderer.createRenderer()
       mapboxRenderer.onSurfaceChanged(1, 1)
     }
   }
 
   @Test
   fun onSurfaceWithActivityDestroyedAfterSurfaceTest() {
-    val latch = CountDownLatch(1)
-    every { mapboxRenderer.destroyRenderer() } answers { latch.countDown() }
-    mockValidSurface()
-    mapboxRenderThread.onSurfaceDestroyed()
-    mapboxRenderThread.destroy()
-    if (!latch.await(waitTime, TimeUnit.MILLISECONDS)) {
-      throw TimeoutException()
+    initRenderThread()
+    provideValidSurface()
+    waitZeroCounter {
```

Comment:
Here and below we need latches since Handler is stopped in destroy().

Other tests are rewritten so that unneeded latches are not created - instead we `pause` and `idle` handler thread accordingly to handle posted runnables